### PR TITLE
[PDR-956] Update and add new workbench tables in PDR

### PR DESCRIPTION
**Pull request by robabram**
## Description of changes/additions
* Added workbench Audit table to PDR
* Switched to using Researcher and Institution Affiliation history tables
* Update bad BigQuery view definitions

## Tests
- [x] unit tests

### CODE REVIEW

File: rdr_service/api/redcap_workbench_audit_api.py [hunk 1]
```diff
@@ -57,6 +59,20 @@ def get(self):
...
         return self.dao.workspace_dao.get_redcap_audit_workspaces(**self.get_filters)
 
+    def _do_insert(self, m):
+        audit_records = super()._do_insert(m)
+        # Generate tasks to build PDR records.
+        if GAE_PROJECT == 'localhost':
+            rebuild_bq_audit(audit_records)
+        else:
+            payload = {'table': 'audit', 'ids': []}
+            for obj in audit_records:
```
Comment (pegbertsch) [inline rdr_service/api/redcap_workbench_audit_api.py:70]: Do we need to limit the number of ids being rebuilt per task?

File: rdr_service/dao/bq_workbench_dao.py [hunk 2]
```diff
@@ -72,8 +72,9 @@ def make_bqrecord(self, src_pk_id, convert_to_enum=False):
             data['income_level'] = str(WorkbenchWorkspaceIncomeLevel(row.income_level))
             data['income_level_id'] = int(WorkbenchWorkspaceIncomeLevel(row.income_level))
 
-            data['access_tier'] = str(WorkbenchWorkspaceAccessTier(row.access_tier))
-            data['access_tier_id'] = int(WorkbenchWorkspaceAccessTier(row.access_tier))
+            if row.access_tier:
```
Comment (pegbertsch) [inline rdr_service/dao/bq_workbench_dao.py:75]: Is there any need to check for the presence of the other `row.*` values above, before trying to cast  as enum values?

File: rdr_service/dao/workbench_dao.py [hunk 3]
```diff
@@ -1169,6 +1172,30 @@ def _insert_history(session, researchers):
...
             history.workbenchInstitutionalAffiliations = affiliations_history
             session.add(history)
+            hist_researchers.append(history)
+        session.flush()
+        session.commit()
+
+        # Generate tasks to build PDR records.
+        if GAE_PROJECT == 'localhost':
+            rebuild_bq_wb_researchers(hist_researchers)
+        else:
```
Comment (pegbertsch) [inline rdr_service/dao/workbench_dao.py:1182]: Do we need to limit how many records are rebuilt per task?

File: rdr_service/model/bq_workbench_workspace.py [hunk 4]
```diff
@@ -169,3 +172,31 @@ class BQRWBWorkspaceUsersView(BQView):
     __viewdescr__ = 'Research Workbench Workspace Users View'
     __pk_id__ = 'id'
     __table__ = BQRWBWorkspaceUsers
+
+
+class BQRWBAuditSchema(BQSchema):
```
Comment (pegbertsch) [inline rdr_service/model/bq_workbench_workspace.py:177]: So we expect to bring over email and notes text unaltered, and move this to a more locked down PostgreSQL schema?

File: rdr_service/dao/workbench_dao.py [hunk 5]
```diff
@@ -1169,6 +1172,30 @@ def _insert_history(session, researchers):
                 affiliations_history.append(affiliation_obj)
             history.workbenchInstitutionalAffiliations = affiliations_history
             session.add(history)
+            hist_researchers.append(history)
+        session.flush()
+        session.commit()
```
Comment (pegbertsch) [inline rdr_service/dao/workbench_dao.py:1177]: Is the flush() already an implicit part of the commit(), if you don't need to access anything in between the two calls?

**Review by pegbertsch — COMMENTED** [review]
Main question is around lack of limits on ids built per task queued.  This may be deliberate if the expectation is these datasets will remain pretty small.

File: rdr_service/dao/workbench_dao.py (hunk 5 above)
Comment (robabram) [reply rdr_service/dao/workbench_dao.py:1177]: I have removed the Flush() call in the new commit.

File: rdr_service/dao/bq_workbench_dao.py (hunk 2 above)
Comment (robabram) [reply rdr_service/dao/bq_workbench_dao.py:75]: No, the other enum fields have a default value of zero.

File: rdr_service/dao/workbench_dao.py (hunk 3 above)
Comment (robabram) [reply rdr_service/dao/workbench_dao.py:1182]: I have added a 'list_hhunks()' for loop to limit batch size.

File: rdr_service/model/bq_workbench_workspace.py (hunk 4 above)
Comment (robabram) [reply rdr_service/model/bq_workbench_workspace.py:177]: The code to change the values to boolean for these fields is in the BQ generator.

File: rdr_service/api/redcap_workbench_audit_api.py (hunk 1 above)
Comment (robabram) [reply rdr_service/api/redcap_workbench_audit_api.py:70]: I have added a 'list_chunks()' for loop to limit batch size.

**Review by pegbertsch — APPROVED** [review]

**Review by dtharpeuno — APPROVED** [review]

**Pull request merged**